src/util/tools.cc

```diff
@@ -48,6 +48,10 @@
 #include <sys/stat.h>
 #include <unistd.h>
 
+#ifdef __HAIKU__
+#define _DEFAULT_SOURCE
```

Comment:
This is interesting. Why is DEFAULT_SOURCE needed?

Reply:
It guards the BSD additions.
https://github.com/haiku/haiku/blob/master/headers/compatibility/bsd/ifaddrs.h#L10

Reply:
Huh. Interesting.

Reply:
Maybe just features.h should be included, idk.
https://github.com/haiku/haiku/blob/master/headers/compatibility/bsd/features.h